File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    static int count = 0;
    public int solution(int[] num_list) {
        int answer = 0;
        
        for(int i = 0; i < num_list.length; i++){
            count = 0;
            answer += calculate(num_list[i]);    
        }
        
        return answer;
    }
    
    public int calculate(int num){
        if(num == 1) {
            return count;
        }else if(num % 2 == 0){
            count++;
            return calculate(num / 2);
        }else {
            count++;
            return calculate((num - 1) / 2);    
        }
    }
}

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// num_list_len은 배열 num_list의 길이입니다.
int solution(int num_list[], size_t num_list_len) {
    int answer = 0;
    int n;
    
    for(int i = 0; i < num_list_len; i++){
        n = num_list[i];
        
        while(n != 1){
            if(n % 2 == 0){
                n /= 2;
            }else {
                n -= 1;
                n /= 2;
            }
            answer++;
        }
    }
    
    return answer;
}